### DISCUSSION TITLE
Testing libraries

### DISCUSSION THREAD

**Discussion by rob-mccann** [home-assistant/frontend#7490] [Other feature requests]
Hey!

I was wondering if you had any thoughts about unit tests for components or automated integration tests via something like Cypress?

Keen to contribute in this area!

**Comment by KTibow** [comment]
Do you mean for the frontend, or for the backend?

**Reply by zsarnett** [reply]
This is the frontend repo :). We have talked about test for a long time. but no one ever did it because its not fun :)

**Reply by KTibow** [reply]
What happened to:
> We use [BrowserStack](https://www.browserstack.com) to test Home Assistant on a large variety of devices.

Is it not actually being tested?

**Reply by zsarnett** [reply]
We don't use it for automated testing. Just use it for testing features that we add to make sure they work on the different browsers

**Reply by rob-mccann** [reply]
The suggestion is to write some automated integration tests first to test the UI is working with core via cypress.
These are separate and in addition to cross-browser tests as the aim is to detect different kinds of issues.

These tests could be run in the pipeline for either repo to check compatibility and regression test.
